PathPlanning/DStarLite/d_star_lite.py

```diff
@@ -62,32 +65,46 @@ def __init__(self, ox: list, oy: list):
         self.y_max = int(abs(max(oy) - self.y_min_world))
         self.obstacles = [Node(x - self.x_min_world, y - self.y_min_world)
                           for x, y in zip(ox, oy)]
+        self.obstacles_xy = np.array(
+            [[obstacle.x, obstacle.y] for obstacle in self.obstacles]
+        )
         self.start = Node(0, 0)
         self.goal = Node(0, 0)
         self.U = list()  # type: ignore
         self.km = 0.0
         self.kold = 0.0
-        self.rhs = list()  # type: ignore
-        self.g = list()  # type: ignore
+        self.rhs = np.full(
+            (self.x_max, self.y_max),
+            float("inf")  # type: ignore
+        )
+        self.g = np.full(
+            (self.x_max, self.y_max),
+            float("inf")  # type: ignore
+        )
         self.detected_obstacles = list()  # type: ignore
```

Comment:
self.dtected_obstacles is still needed?

Reply:
It is still used in random object generation. I will change that function as well and test it.

Reply:
I removed it, and while doing that fixed a bug about going out of range of the rhs.